### ISSUE TITLE
authorization on push

### ISSUE THREAD

**Issue by ugenk** [arut/nginx-rtmp-module#688]
Is it possible to add login and password to push url? 

I.e. to make something like:
application app00 {
 live on;
 push rtmp://user:pass@server:port/app/;
}

**Comment by deluxor** [comment]
Hello friend, you can use the HTTP callbacks, using the on_publish notify, you just need to stripe the arguments you pass on the connection string and validate them against a PHP

**Comment by ugenk** [comment]
No, i need to authorize on remote rtmp server when doing push

**Comment by misiek08** [comment]
Did you try solution you wrote in question? It should work with auth data
in rtmp url.

2015-09-21 21:11 GMT+02:00 ugenk <notifications@github.com>:

**Comment by ugenk** [comment]
root@gate:~# /usr/local/nginx/sbin/nginx -s reload
nginx: [emerg] invalid port in url "user:pass@1.2.3.4:1935/live/" in /usr/local/nginx/conf/nginx.conf:26

part from config:
		push rtmp://user:pass@1.2.3.4:1935/live/;

**Comment by losywee** [comment]
You can view: http://nginx-rtmp.blogspot.jp/2013/06/secure-links-in-nginx-rtmp.html

**Comment by sgdc3** [comment]
> rtmp://user:pass@1.2.3.4:1935/live/

That syntax would be perfect, @arut any news?

**Comment by eelbaz** [comment]
@arut as @sgdc3  mentions people needing to leverage push rtmp://user:pass@1.2.3.4:1935/live/; to external services that use authenticated rtmp, nginx-rtmp does not work. Can you help shed some light on how to accomplish rtmp://user:pass@1.2.3.4:1935/live/ with out getting the "invalid port in url" error?  

Thanks in Advance!

**Comment by crowdoptic** [comment]
+1 We also need it to push to Wowza

**Comment by paulgalow** [comment]
+1 That would be really helpful.

**Comment by shahab1363** [comment]
I tried ffmpeg and it worked like a charm (in config file):
exec ffmpeg -i rtmp://127.0.0.1/live/test -threads 1 -vcodec copy -acodec copy -f flv rtmp://[username]:[password]@dacast.stream.target.url

**Comment by sgdc3** [comment]
@shahab1363 nice! thank you for sharing

**Comment by renatosbc** [comment]
+1. This  would be really helpful.

**Comment by shahab1363** [comment]
@renatosbc you are right, but cpu load was too low in real world scenario (around 1% in a virtual machine with very limited resources - tested for more than three months).
I made a small bash script to restart ffmpeg on errors:
`#!/bin/bash

while true
do
    ffmpeg -i rtmp://127.0.0.1/live/test -threads 1 -vcodec copy -acodec copy -f flv rtmp://[username]:[password]@dacast.stream.target.url -xerror 2> /tmp/ffmpeg.log
    sleep 1
done`

and updated the config file to run it:
`application live {
                 live on;
                 record off;
                 exec /path/to/shell/script.sh;
}`

@renatosbc @sgdc3 you are very welcome.
